src/Voron/StorageEnvironmentOptions.cs

```diff
@@ -471,6 +473,20 @@ protected override void Disposing()
                     if (journal.Value.IsValueCreated)
                         journal.Value.Value.Dispose();
                 }
+
+                lock (_journalsForReuse)
+                {
+                    foreach (var reusableFile in _journalsForReuse.Values)
```

Comment:
I'm for the approach that the recycle areas (for scratches and journals) exist only when a db is live. Previously we were not consistent here - recycled scratches were deleted while scratches were not.